### ISSUE TITLE
[BUG - Develop] Exponential growth in EvTable columns

### ISSUE THREAD

**Issue by owllex** [evennia/evennia#2653]
#### Describe the bug

Repeated called to `add_row()` on an EvTable with an uneven number of columns causes the table to start doubling its column count with subsequent calls. Eventually the table grows so big that memory operations to expand the table causes those calls to start taking many seconds each.

#### To Reproduce

Here is a minimal test case that reliably reproduces the problem:
```py
from django.test import SimpleTestCase
from evennia.utils.evtable import EvTable

class TestTable(SimpleTestCase):
    def test(self):
        # Note: set range to 50+ to see your computer lock up.
        table = EvTable(border=None)
        for i in range(35):
            if i % 5 == 0:
                table.add_row("Single")
            else:
                table.add_row("Double", "Double")
            print(f"Iteration {i}: cols: {table.ncols}, rows: {table.nrows}")
        print(str(table))
```

1. Add above unit test and run.
2. The test eventually times out.
3. Observe output:
Iteration 0: cols: 1, rows: 1
Iteration 1: cols: 2, rows: 2
Iteration 2: cols: 2, rows: 3
Iteration 3: cols: 2, rows: 4
Iteration 4: cols: 2, rows: 5
Iteration 5: cols: 3, rows: 6
Iteration 6: cols: 4, rows: 7
Iteration 7: cols: 6, rows: 8
Iteration 8: cols: 10, rows: 9
Iteration 9: cols: 18, rows: 10
Iteration 10: cols: 35, rows: 11
Iteration 11: cols: 68, rows: 12
Iteration 12: cols: 134, rows: 13
Iteration 13: cols: 266, rows: 14
Iteration 14: cols: 530, rows: 15
Iteration 15: cols: 1059, rows: 16
Iteration 16: cols: 2116, rows: 17
Iteration 17: cols: 4230, rows: 18
Iteration 18: cols: 8458, rows: 19
Iteration 19: cols: 16914, rows: 20
Iteration 20: cols: 33827, rows: 21
Iteration 21: cols: 67652, rows: 22
Iteration 22: cols: 135302, rows: 23
Iteration 23: cols: 270602, rows: 24
Iteration 24: cols: 541202, rows: 25
Iteration 25: cols: 1082403, rows: 26
Iteration 26: cols: 2164804, rows: 27
Iteration 27: cols: 4329606, rows: 28
Iteration 28: cols: 8659210, rows: 29
Iteration 29: cols: 17318418, rows: 30
Iteration 30: cols: 34636835, rows: 31
Iteration 31: cols: 69273668, rows: 32
Iteration 32: cols: 138547334, rows: 33
Iteration 33: cols: 277094666, rows: 34
Iteration 34: cols: 554189330, rows: 35

#### Expected behavior

I would have expected the table to only ever have 2 columns, as each row is no more than that in length.

#### Develop-branch commit
    
    Evennia 1.0-dev (rev 2e645c58b) (rev 2e645c58b)
    OS: posix
    Python: 3.10.2
    Twisted: 21.7.0
    Django: 4.0.2

#### Additional context

I instrumented the code a bit to see what was going on, and at some point, a mysterious new column has appeared. Everything being added to the table is a fixed width, so it's unclear why this happens. Once it does, reliably around row 7, each subsequent call looks like the table is missing one or more columns and starts adding them, causing the exponential growth.

**Comment by owllex** [comment]
The issue does not occur if you modify `add_row()` to always supply the same number of columns. So in the above test, changing the line to `add_row("Single", "")` has the same visual effect but does not trigger the exponential growth in columns.

**Comment by Griatch** [comment]
Kudos for providing a unit test - this made this easy to reproduce! :+1: After some digging, I figured out what was going on here - the alternating row lengths confused the algorithm. Fixed now and 35 rows are now added in 0.07s.
